lib/request_handler/schema_parser.rb

```diff
@@ -27,9 +27,13 @@ def validate(data, schema:)
     end
 
     def validation_failure?(validator)
-      return if validator.valid?
+      if validator.respond_to?(:valid?)
+        return if validator.valid?
+      elsif validator.success?
+        return
+      end
```

Comment:
is this a weird dry thing or why is there sometimes no `valid?` function?

Reply:
Thanks! That's not needed anymore. Removed it!